18F/identity-idp
# LG-10022: Add second MFA reminder screen

## 🎫 Ticket

[LG-10022](https://cm-jira.usa.gov/browse/LG-10022)

## 🛠 Summary of changes

Adds a new MFA reminder screen that shows if the user only has a single MFA method, upon the first of either (a) the user has signed in 10 or more times or (b) their account is 30 days old.

## 📜 Testing Plan

It's easiest to test by setting an artificially low config override for sign-ins or account page, e.g. in `config/application.yml`:

```
second_mfa_reminder_sign_in_count: 2
```

1. Go to http://localhost:3000
2. Create an account
3. After account creation, sign out
4. Sign in
5. Observe that you are prompted with the screen in screenshots below if (and only if) you had set up a single method during account creation
6. Click one of the action buttons
7. Observe that:
   a. If you chose to set up another MFA method, you're redirected to the MFA setup screen, with a "Cancel" link at the bottom
   b. If you chose to Continue, you're redirected to your account dashboard (or to the partner if coming from a partner authentication request)
8. Sign out
9. Sign in
10. Observe that you are not prompted with the screen in screenshots below regardless of which action you took at Step 6

## 👀 Screenshots

![Screenshot 2023-08-31 at 10 22 25 AM](https://github.com/18F/identity-idp/assets/1779930/8e055ce6-a16f-41f3-a6fb-4253ba74a07c)
